django/contrib/postgres/indexes.py

```diff
@@ -43,5 +43,52 @@ def get_sql_create_template_values(self, model, schema_editor, using):
 class GinIndex(Index):
     suffix = 'gin'
 
+    GIN_PENDING_LIST_LOWER = 2**6
+    GIN_PENDING_LIST_UPPER = 2**31 - 1
+
+    def __init__(self, fields=[], name=None, fastupdate=None, gin_pending_list_limit=None):
+        if gin_pending_list_limit is not None and not (
+                self.GIN_PENDING_LIST_LOWER <= gin_pending_list_limit <= self.GIN_PENDING_LIST_UPPER):
+            raise ValueError('gin_pending_list_limit must be between {} and {}'.format(
+                self.GIN_PENDING_LIST_LOWER, self.GIN_PENDING_LIST_UPPER))
+        self.fastupdate = fastupdate
+        self.gin_pending_list_limit = gin_pending_list_limit
+        super().__init__(fields, name)
+
+    def deconstruct(self):
+        path, args, kwargs = super().deconstruct()
+        kwargs['fastupdate'] = self.fastupdate
+        kwargs['gin_pending_list_limit'] = self.gin_pending_list_limit
+        return path, args, kwargs
+
+    def get_sql_create_template_values(self, model, schema_editor, using):
+        parameters = super().get_sql_create_template_values(model, schema_editor, using=' USING gin')
+        with_params = []
+        if self.gin_pending_list_limit is not None:
+            with_params.append('gin_pending_list_limit = %d' % self.gin_pending_list_limit)
+        if self.fastupdate is not None:
+            with_params.append('fastupdate = %s' % 'on' if self.fastupdate else 'off')
+        if with_params:
+            parameters['extra'] = 'WITH ({}) {}'.format(
+                ', '.join(with_params), parameters['extra'])
+        return parameters
+
+    def __repr__(self):
```

Comment:
OK. I doubt someone would configure an index for the same columns with different values for those two parameters. The default values chosen for them are probably based on good assumptions for the average user.